arrow-fx-coroutines/src/main/kotlin/arrow/fx/coroutines/timer.kt

```diff
@@ -37,6 +39,32 @@ suspend fun sleep(duration: Duration): Unit =
  * }
  * ```
  **/
-@Deprecated("use withTimeOutOrNull", ReplaceWith("withTimeoutOrNull(duration.millis)", "kotlinx.coroutines.withTimeoutOrNull"))
+@Deprecated(
+  "use withTimeOutOrNull or timeOutOrNull(kotlin.time.Duration, fa)",
+  ReplaceWith("withTimeoutOrNull(duration.millis)", "kotlinx.coroutines.withTimeoutOrNull")
+)
 suspend fun <A> timeOutOrNull(duration: Duration, fa: suspend () -> A): A? =
   withTimeoutOrNull(duration.millis) { fa.invoke() }
+
+/**
+ * Convenience function that returns the result of [fa] within the specified [duration] or returns null.
+ *
+ * ```kotlin:ank:playground
+ * import kotlin.time.seconds
+ * import kotlinx.coroutines.delay
+ * import arrow.fx.coroutines.*
+ *
+ * suspend fun main(): Unit {
+ *   timeOutOrNull(2.seconds) {
+ *     delay(5.seconds)
+ *     "Message from lazy task"
+ *   }.also(::println)
+ *
+ *   timeOutOrNull(2.seconds) {
+ *     "Message from fast task"
+ *   }.also(::println)
+ * }
+ * ```
+ **/
+suspend fun <A> timeOutOrNull(duration: kotlin.time.Duration, fa: suspend () -> A): A? =
```

Comment:
Users should migrate to KotlinX `withTimeoutOrNull`.

Reply:
I will remove that :D